torch/_inductor/ir.py

```diff
@@ -4232,15 +4232,23 @@ def set_indirect(new_var):
                 )
                 return var
 
+            @staticmethod
+            def output(result):
+                tracer.create_proxy("output", "output", (result,), {})
+
         tracer = torch.fx.Tracer()
         tracer.graph = torch.fx.Graph(tracer_cls=tracer.__class__)
         proxy_ops = tracer.create_proxy("placeholder", "ops", (), {})
+
+        from .index_propagation import IndexPropagation
         from .sizevars import SimplifyIndexing
 
         with V.set_ops_handler(
-            SimplifyIndexing(CaptureIndexing(proxy_ops), self.body.var_ranges)
+            IndexPropagation(
+                SimplifyIndexing(CaptureIndexing(proxy_ops), self.body.var_ranges)
+            )
         ):
-            tracer.create_proxy("output", "output", (fn(*args),), {})
+            ops.output(fn(*args))
```

Comment:
This indirection is just a cute way to get `IndexPropagation` to unwrap the return value.